### PR TITLE
Use infinite-scrolling on lists

### PR DESCRIPTION
**Pull request by bentiss**
This allows to reduce the loading time of a big board

Based on http://www.meteorpedia.com/read/Infinite_Scrolling

Related to #2104 

Any tests before merge would be appreciated :)

Note: Apache I-CLA signed

---
This change is [Reviewable](https://reviewable.io/reviews/wekan/wekan/2144)

### CODE REVIEW

**Comment by bentiss** [comment]
.... and if the list has more than 10 cards but they do not fill up the entire container (i.e. no scrolling bar available), there is no way of retrieving the rest of the cards but resizing the window. :(

@xet7: please do not merge it just yet until this is fixed (I won't be able to fix it today IMO)

**Comment by xet7** [comment]
Ok

**Comment by bentiss** [comment]
@xet7 
I just forced push the branch. It should be better now:
- there is already infinite scrolling in the wekan code for the activities list. However, using this code has not been a good experience here given the constraints of the cards (variable height mostly)
- I relied on the idea of checking if the spinner was shown on the list to trigger the next page. This seems to be working quite well.
- 'add a new card' also makes sure the card limit gets increased when necessary
- the code to update the card limit is run when the window gets resized, or we might end up in situation where the spinner stays here without any way of expanding the list of cards.

**Comment by xet7** [comment]
@bentiss 

In Lists and Swimlane mode, when I'm writing text to minicard, pressing Enter duplicates the same text to next card.

Pressing Enter should add card below, and go to card below, without duplicating text.

Pressing Shift-Enter should go to next line on current card.

File: client/components/lists/listBody.js [hunk 1]
```diff
@@ -60,6 +88,13 @@ BlazeComponent.extendComponent({
         type: 'cardType-card',
       });
 
+      // if the displayed card count is less than the total cards in the list,
+      // we need to increment the displayed card count to prevent the spinner
+      // to appear
+      const cardCount = this.data().cards(this.idOrNull(swimlaneId)).count();
+      if (pthis.cardlimit.get() < cardCount) {
```
Comment (bentiss) [inline client/components/lists/listBody.js]: > In Lists and Swimlane mode, when I'm writing text to minicard, pressing Enter duplicates the same text to next card.
> 
> Pressing Enter should add card below, and go to card below, without duplicating text.

yep, typo: `pthis.cardlimit.get()` should be `this.cardlimit.get()`

This must have chimed in while I updated the comment above, sorry.

**Comment by bentiss** [comment]
> In Lists and Swimlane mode, when I'm writing text to minicard, pressing Enter duplicates the same text to next card.
> 
> Pressing Enter should add card below, and go to card below, without duplicating text.

thanks for spotting this, fixed.

> Pressing Shift-Enter should go to next line on current card.

This is not the current behavior on `edge`. Let me see if I can find the culprit.

**Comment by bentiss** [comment]
>>  Pressing Shift-Enter should go to next line on current card.
>>
> This is not the current behavior on edge. Let me see if I can find the culprit.

This should be working (but this is not related to this PR):

```diff
diff --git a/client/components/lists/listBody.js b/client/components/lists/listBody.js
index 1001f3bc..5d325898 100644
--- a/client/components/lists/listBody.js
+++ b/client/components/lists/listBody.js
@@ -179,7 +179,7 @@ BlazeComponent.extendComponent({
 
   pressKey(evt) {
     // Pressing Enter should submit the card
-    if (evt.keyCode === 13) {
+    if (evt.keyCode === 13 && !evt.shiftKey) {
       evt.preventDefault();
       const $form = $(evt.currentTarget).closest('form');
       // XXX For some reason $form.submit() does not work (it's probably a bug
```

**Pull request merged**